lxd/db/certificates.go

```diff
@@ -103,6 +103,11 @@ func (cert *Certificate) ToAPI() api.Certificate {
 
 // UpdateCertificateProjects updates the list of projects on a certificate.
 func (c *ClusterTx) UpdateCertificateProjects(id int, projects []string) error {
+	// If there are no project names, there's nothing to do.
+	if len(projects) == 0 {
+		return nil
```

Comment:
Shouldn't this case either remove all entries for that certificate, or return an error?